index.js

```diff
@@ -52,4 +52,129 @@ const state = {
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+
+// Select the store ul, cart ul and total cost elements
+const storeUl = document.querySelector('.store--item-list')
+const cartUl = document.querySelector('.cart--item-list')
+const totalNum = document.querySelector('.total-number')
+
+// Use a for loop to cycle through the state item
+// and create the icons and buttons for each.
+
+for (let i = 0; i < state.items.length; i++) {
+
+  // create elements
+  const storeLi = document.createElement('li')
+  const storeImg = document.createElement('img')
+  const storeBtn = document.createElement('button')
+
+  // add a source to the img using the created variable
+  // add a class to the img to apply correct styling
+  storeImg.setAttribute('src', `assets/icons/${state.items[i].id}.svg`)
+  storeImg.setAttribute('class', '.store--item-icon')
+
+  // add the text to the button
+  storeBtn.innerText = 'Add to cart'
+
+  // add event listener to button which checks if
+  // the item is not in the cart ! (bang). If not then
+  // it will add the item while giving it a quantity
+  // of one. If it is already in the cart then
+  // the item quantity will increase by 1
+  storeBtn.addEventListener('click', () => {
+    if (!state.cart.includes(state.items[i])) {
```

Comment:
Very nice workflow, Lee. Compared to my code, I think I will prefer your workflow. Well done. The only thing I will suggest is that you should try not to mutate the state.items array. So, instead of adding  the quantity field inside the items array, you should probably add it to the cart array. My reason is that quantity is not really required inside the items array.

Reply:
Yes I'm sure I had this to begin with, The reason I had it here is because I wanted the quantity to change when 'add to cart' was clicked more than once